jmxtrans-output/jmxtrans-output-elastic/src/main/java/com/googlecode/jmxtrans/model/output/elastic/ElasticWriter.java

```diff
@@ -0,0 +1,219 @@
+public class ElasticWriter extends BaseOutputWriter {
+	
+	private static final Logger log = LoggerFactory.getLogger(ElasticWriter.class);
+	
+	private static final String DEFAULT_ROOT_PREFIX = "jmxtrans";
+	private static final Object CREATE_MAPPING_LOCK = new Object();
+
+	private final JsonFactory jsonFactory;
+
+	private JestClient jestClient;
+
+	private final String rootPrefix;
+	private final String connectionUrl;
+	private static final String TYPE_NAME = "jmx-entry";
+	private String indexName;
+
+	@JsonCreator
+	public ElasticWriter(
+			@JsonProperty("typeNames") ImmutableList<String> typeNames,
+			@JsonProperty("booleanAsNumber") boolean booleanAsNumber,
+			@JsonProperty("rootPrefix") String rootPrefix,
+			@JsonProperty("debug") Boolean debugEnabled,
+			@JsonProperty("connectionUrl") String connectionUrl,
+			@JsonProperty("settings") Map<String, Object> settings) throws IOException {
+
+		super(typeNames, booleanAsNumber, debugEnabled, settings);
+
+		this.rootPrefix = resolveProps(
+				firstNonNull(
+						rootPrefix,
+						(String) getSettings().get("rootPrefix"),
+						DEFAULT_ROOT_PREFIX));		
+
+		this.jsonFactory = new JsonFactory();
+		this.connectionUrl = connectionUrl;
+		this.indexName = this.rootPrefix + "_jmx-entries";
+		this.jestClient = createJestClient(connectionUrl);
+	}
+
+	private JestClient createJestClient(String connectionUrl) {
+		log.info("Create a jest elastic search client for connection url [{}]", connectionUrl);
+		JestClientFactory factory = new JestClientFactory();
+		factory.setHttpClientConfig(
+				new HttpClientConfig.Builder(connectionUrl)
+						.multiThreaded(true)
+						.build());
+		return factory.getObject();
+	}
+
+	@Override
+	protected void internalWrite(Server server, Query query, ImmutableList<Result> results) throws Exception {
+		List<String> typeNames = this.getTypeNames();
+
+		for (Result result : results) {
+			log.debug("Query result: [{}]", result);
+			Map<String, Object> resultValues = result.getValues();
+			for (Entry<String, Object> values : resultValues.entrySet()) {
+				Object value = values.getValue();
+				if (isNumeric(value)) {
+					String message = createJsonMessage(server, query, typeNames, result, values, value);
+					log.debug("Insert into Elastic: Index: [{}] Type: [{}] Message: [{}]", indexName, TYPE_NAME, message);
+					Index index = new Index.Builder(message).index(indexName).type(TYPE_NAME).build();
+					jestClient.execute(index);
+				} else {
+					log.warn("Unable to submit non-numeric value to Elastic: [{}] from result [{}]", value, result);
+				}
+			}
+		}
+	}
+
+	private String createJsonMessage(Server server, Query query, List<String> typeNames, Result result, Entry<String, Object> values, Object value) throws IOException {
+
+		String keyString = getKeyString(server, query, result, values, typeNames, this.rootPrefix);
+
+		Closer closer = Closer.create();
+		try {
+			ByteArrayOutputStream out = closer.register(new ByteArrayOutputStream());
```

Comment:
It seems that Jest supports getting data from a simple `Map`. Is there any reason to use JSON ? (That's an honest question, the map seems simpler to me, but you probably have a reason for going to JSON ...)